src/main/java/decodes/cwms/CwmsSqlDatabaseIO.java

```diff
@@ -179,114 +193,113 @@ public void connectToDatabase(String sqlDbLocation)
-			Logger.instance().failure(msg);
-			System.err.println(msg);
-			ex.printStackTrace(System.err);
-			cgl.setLoginSuccess(false);
-			if (conInfo.getConnection() != null)
-				freeConnection(conInfo.getConnection());
-			conInfo = null;
-			throw new DatabaseException(msg);
+			conInfo = new CwmsConnectionInfo();
+			ConnectionLoginInfo info = new ConnectionLoginInfoImpl(sqlDbLocation, _dbUser,password, dbOfficeId);
+			conInfo.setLoginInfo(info);
 		}
-		
-		readVersionInfo(this, conInfo.getConnection());
-		
-		// CWMS OPENDCS-16 for DB version >= 68, use old OracleSequenceKeyGenerator,
-		// which assumes a separate sequence for each table. Do not use CWMS_SEQ for anything.
-		int decodesDbVersion = getDecodesDatabaseVersion();
-		Logger.instance().info(module + " decodesDbVersion=" + decodesDbVersion);
-		keyGenerator = decodesDbVersion >= DecodesDatabaseVersion.DECODES_DB_68 ?
-				new OracleSequenceKeyGenerator() :
-				new CwmsSequenceKeyGenerator(decodesDbVersion);
 
-		String q = null;
-		Statement st = null;
 		try
 		{
-			st = conInfo.getConnection().createStatement();
-			
-			// Hard-code date & timestamp format for reads. Always use GMT.
-			q = "ALTER SESSION SET TIME_ZONE = 'GMT'";
-			Logger.instance().info(q);
-			st.execute(q);
+			pool = CwmsConnectionPool.getPoolFor(conInfo);
+			if( pool != null)
+			{
+				try(Connection conn = pool.getConnection();)
+				{
+					Logger.instance().info(module +
+						" Connected to DECODES CWMS Database " + sqlDbLocation + " as user " + _dbUser
+						+ " with officeID=" + dbOfficeId);
+					readVersionInfo(this, conn);
 
-			q = "ALTER SESSION SET nls_date_format = 'yyyy-mm-dd hh24:mi:ss'";
-			Logger.instance().info(q);
-			st.execute(q);
-			
-			q = "ALTER SESSION SET nls_timestamp_format = 'yyyy-mm-dd hh24:mi:ss'";
-			Logger.instance().info(q);
-			st.execute(q);
+					// CWMS OPENDCS-16 for DB version >= 68, use old OracleSequenceKeyGenerator,
+					// which assumes a separate sequence for each table. Do not use CWMS_SEQ for anything.
+					int decodesDbVersion = getDecodesDatabaseVersion();
+					Logger.instance().info(module + " decodesDbVersion=" + decodesDbVersion);
+					keyGenerator = decodesDbVersion >= DecodesDatabaseVersion.DECODES_DB_68 ?
+							new OracleSequenceKeyGenerator() :
+							new CwmsSequenceKeyGenerator(decodesDbVersion);
+						TimeSeriesDb.readVersionInfo(this, conInfo.getConnection());
+						cgl.setLoginSuccess(true);
 
-			q = "ALTER SESSION SET nls_timestamp_tz_format = 'yyyy-mm-dd hh24:mi:ss'";
-			Logger.instance().info(q);
-			st.execute(q);
-			
-			Logger.instance().info("DECODES IF Connected to TSDB Version " + tsdbVersion);
-		}
-		catch(SQLException ex)
-		{
-			String msg = "Error in '" + q + "': " + ex
-				+ " -- will proceed anyway.";
-			Logger.instance().failure(msg + " " + ex);
-		}
-		finally
-		{
-			try { st.close(); } catch(Exception ex) {}
-		}
+					String q = null;
+					try(Statement st = conn.createStatement();)
```

Comment:
recommend moving every statement into its own method to help debug connection management, as well as keeping the scope that is using the connection as minimal as possible.

Reply:
agreed, but slightly beyond scope of this PR.